scene/realistic-lighting-and-shadows/src/main/java/com/esri/samples/realistic_lighting_and_shadows/RealisticLightingAndShadowsController.java

```diff
@@ -0,0 +1,217 @@
+
+      // set atmosphere effect to realistic
+      sceneView.setAtmosphereEffect(AtmosphereEffect.REALISTIC);
+
+      // set a new calendar and add a date and time
+      calendar = new GregorianCalendar(2018, 7, 10, 12, 00, 0);
+      calendar.setTimeZone(TimeZone.getTimeZone("PST"));
+
+      // set the time label on the control panel
+      sceneView.setSunTime(calendar);
+
+      // tidy string to just return date and time (hours and minutes)
+      dateFormat = new SimpleDateFormat("EEE MMM dd HH:mm");
+      dateFormat.setTimeZone(TimeZone.getTimeZone("PST"));
+      String dateAndTimeTidied = dateFormat.format(calendar.getTime());
+
+      // set a label to display the tidied date and time
+      time.setText(dateAndTimeTidied);
+
+      // set the slider to display tick labels as time strings
+      setSliderLabels();
+
+      // set the combo box to display the lighting modes as strings
+      setComboBoxLabels();
+
+      // update the atmosphere effect based on the lighting mode chosen from the combo box
+      comboBox.getSelectionModel().selectedItemProperty().addListener(e -> {
+        sceneView.setSunLighting(comboBox.getSelectionModel().getSelectedItem());
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Set the sun time based on the time from the slider.
+   */
+  @FXML
+  public void changeTimeOfDay() {
+    // when the slider changes, update the hour of the day based on the value of the slider
+    timeSlider.valueProperty().addListener((observable, oldValue, newValue) -> {
+
+        // get the hour value from the slider
+        int hours = newValue.intValue();
+
+        // get the minutes from the slider
+        double afterDecimal = newValue.doubleValue() - floor(newValue.doubleValue());
+        int minutes = (int) (afterDecimal * 60);
+
+        // set the calendar for given hour and minute from slider value
+        calendar.set(Calendar.HOUR_OF_DAY, hours);
+        calendar.set(Calendar.MINUTE, minutes);
+
+        // tidy string to just return date and time (hours and minutes)
+        String dateAndTimeTidied = dateFormat.format(calendar.getTime());
+
+        // update label to reflect current date and time
+        time.setText(dateAndTimeTidied);
+
+        // set the sun time to calendar
+        sceneView.setSunTime(calendar);
+      }
+    );
+  }
+
+  /**
+   * Set labels to display on the slider.
+   */
+  private void setSliderLabels() {
+
+    timeSlider.setLabelFormatter(new StringConverter<Double>() {
+
+      @Override
+      public String toString(Double hour) {
+
+        if (hour == 4) return "4am";
+        if (hour == 8) return "8am";
+        if (hour == 12) return "Midday";
+        if (hour == 16) return "4pm";
+        if (hour == 20) return "8pm";
+
+        return "Midnight";
+      }
+
+      @Override
+      public Double fromString(String string) {
+        return null;
+      }
+    });
```

Comment:
On second thought, we can also extract this new class definition for StringConverter into a separate file, then just do
`timeSlider.setLabelFormatter(new CustomStringConverter());`
That way we can leave this method out and have a single line in the main `initialize()` method.

My thought behind this is that it's more JavaFX/UI setup, and our main goal with the Java samples is to teach people the main concepts of the Runtime API. In this case the 'meat' of this sample is your `changeTimeOfDay()` method so we want to bring that front and centre